memif-attach/main.go

```diff
@@ -18,59 +18,102 @@ package main
 
 import (
 	"context"
+	"net"
 	"time"
-
+	// "io"
+	"encoding/binary"
+	"bufio"
+	"bytes"
 	"git.fd.io/govpp.git/api"
 	"github.com/edwarnicke/govpp/binapi/memif"
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/edwarnicke/govpp/binapi/interface_types"
 	"github.com/edwarnicke/log"
 	"github.com/edwarnicke/vpphelper"
 )
 
+// HelloReplyMsg type
+type HelloMsg struct {
+	max_region int16 
+	max_m2s_ring int16 
+	max_s2m_ring int16
+	min_version int16 
+	max_version int16
+	max_log2_ring_size int8 
```

Comment:
Not that this is the source of your current problem, but these are likely supposed to be unsigned, check the original info in the vpp source code to be sure.

Reply:
@harshgondaliya pointed you that this is because your fields in your struct are private, and so reflect cannot access them.

This is actually an interesting point about Go.  Go makes capitalized things public.  So the type HelloMsg is public, because it is capitalized.  But its field HellMsg.max_region is private, because max_region is capitalized, so it cannot be read out from outside your current package.  If you capitalize all the fields they should be accessible.

https://stackoverflow.com/questions/53293862/panic-setuint-using-value-obtained-using-unexported-field